### DISCUSSION TITLE
How to add increment primary key in an existing PostgreSQL table?

### DISCUSSION THREAD

**Discussion by DesmondSol** [hasura/graphql-engine#8784] [Question] [answered]
when creating my tables in the hasura console i forgot to make the id auto incremental for all the tables i used and after that when i tried to change them all to that there wasn't any easy way available i could find so i had to create alter statements for all the tables on the raw sql to do my tasks but since that is a lot of work can i ask if there is a better way already made.
 
i used this code to make every change for all 35 tables that i had
```
CREATE SEQUENCE winning_bids_details_id_seq OWNED BY winning_bids_details.id;

ALTER TABLE winning_bids_details ALTER COLUMN id SET DEFAULT nextval('winning_bids_details_id_seq');

UPDATE
	winning_bids_details
SET
	id = nextval('winning_bids_details_id_seq');
```

**Comment by adas98012** [comment, marked as answer]
Please see if the following discussion thread is helpful to your case:

https://dba.stackexchange.com/questions/285869/add-auto-increment-to-already-existing-primary-key-column-postgresql

**Reply by DesmondSol** [reply]
ya i see but if there was just a selection available in the hasura to add this for each column edit it would have been awesome but thanks for your time
